tests/flamegraph.rs

```diff
@@ -116,9 +167,128 @@ fn flamegraph_nameattr() {
         hash: true,
         func_frameattrs: flamegraph::FuncFrameAttrsMap::from_file(&PathBuf::from(nameattr_file))
             .unwrap(),
-        title: "Flame Graph".to_string(),
         ..Default::default()
     };
 
-    test_flamegraph(input_file, expected_result_file, options);
+    test_flamegraph(input_file, expected_result_file, options).unwrap();
+}
+
+#[test]
+fn flamegraph_should_warn_about_fractional_samples() {
+    test_flamegraph_logs(
+        "./tests/data/fractional-samples/fractional.txt",
+        |captured_logs| {
+            assert_eq!(captured_logs.len(), 1);
+            assert!(captured_logs[0].body.contains("fractional sample"));
+            assert_eq!(captured_logs[0].level, Level::Warn);
+        },
+    );
+}
+
+#[test]
+fn flamegraph_should_not_warn_about_zero_fractional_samples() {
+    test_flamegraph_logs(
+        "./tests/data/fractional-samples/zero-fractionals.txt",
+        |captured_logs| {
+            assert!(captured_logs.is_empty());
+        },
```

Comment:
Here too, I think we probably want to be a little more liberal. Specifically, check that there are no messages about fractional numbers.